### PR TITLE
remove old MergeAnnotation func and rename ReplaceAnnotation to MergeAnnotation

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Learn from #4751, to avoid ambiguity and to be consistent with the Annotation naming convention:

https://github.com/karmada-io/karmada/blob/57c1989667d826864aa09105daa04434e2c75dd9/pkg/util/annotation.go#L29-L51

remove old MergeAnnotation func and rename ReplaceAnnotation to MergeAnnotation.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by whitewindmills — COMMENTED** [review]
/lgtm

File: pkg/controllers/binding/common.go [hunk 1]
```diff
@@ -154,18 +154,18 @@ func mergeTargetClusters(targetClusters []workv1alpha2.TargetCluster, requiredBy
 
 func mergeLabel(workload *unstructured.Unstructured, binding metav1.Object, scope apiextensionsv1.ResourceScope) map[string]string {
 	var workLabel = make(map[string]string)
-	util.MergeLabel(workload, util.ManagedByKarmadaLabel, util.ManagedByKarmadaLabelValue)
+	util.ReplaceLabel(workload, util.ManagedByKarmadaLabel, util.ManagedByKarmadaLabelValue)
```
Comment (RainbowMango) [inline pkg/controllers/binding/common.go:157]: fells not right, as the workload might not have the label, how does `replace`?

Comment (XiShanYongYe-Chang) [reply pkg/controllers/binding/common.go:157]: Any suggestion?

Comment (XiShanYongYe-Chang) [reply pkg/controllers/binding/common.go:157]: I'm thinking that MergeLabel is the  appropriate name. The original MergeAnnotation function may not be necessary:

https://github.com/karmada-io/karmada/blob/4e2548250b19a8f8148c004defa02b0e99f5a3e7/pkg/util/annotation.go#L29-L40

File: pkg/util/annotation_test.go [hunk 2]
```diff
@@ -25,118 +25,6 @@ import (
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
 )
 
-func TestMergeAnnotation(t *testing.T) {
```
Comment (RainbowMango) [inline pkg/util/annotation_test.go:28]: Why remove this test?

Comment (XiShanYongYe-Chang) [reply pkg/util/annotation_test.go:28]: When ReplaceAnnotation is replaced with MergeAnnotation, the test method name is not changed. Just update.

**Comment by XiShanYongYe-Chang** [comment]
Can you help take a look again @whitewindmills @RainbowMango ?

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**